packages/nuxt-module/plugins/domain.js

```diff
@@ -50,8 +51,19 @@ export default async ({ app, route, req }, inject) => {
         hostname = window.location.hostname;
         pathname = window.location.pathname;
       } else if (!process.static) {
-        const detectedHost =
-          req.headers["x-forwarded-host"] || req.headers.host;
+        // for ALB healtchecks we simply take the first available domain, so we dont generate errors
+        if (req.headers['user-agent'].includes('ELB-HealthChecker') && domainsList.length > 0) {
+          return domainsList[0]
+        }
+
+        let detectedHost = req.headers["x-forwarded-host"] || req.headers.host;
+        // also check the Forwarded header (RFC 7239)
+        if (req.headers["forwarded"]) {
+          const parsedHeader = forwardedParse(req.headers["forwarded"]);
```

Comment:
according to the documentation `forwardedParse` function returns an array of objects, not an object. 
so please add a validation and appropriate data extraction, like `parsedHeader?.[0]?.host or filter an array by allowed domain list (utilize `domainsList` variable for that) and then get the host property's value.

Reply:
fixed it, as there is only 1 forwarded header, i guess `parsedHeader?.[0]?.host` is perfectly fine

Reply:
I agree, it's a good starting point of making it more useful in production environment 👍🏻